server/src/test/java/org/elasticsearch/common/blobstore/fs/FsBlobContainerTests.java

```diff
@@ -165,6 +167,35 @@ public void testCompareAndExchange() throws Exception {
         );
     }
 
+    public void testAtomicWriteMetadataWithoutAtomicOverwrite() throws IOException {
+        this.fileSystem = new FilterFileSystemProvider("nooverwritefs://", fileSystem) {
+            @Override
+            public void move(Path source, Path target, CopyOption... options) throws IOException {
+                final Set<CopyOption> optionSet = Set.of(options);
+                if (optionSet.contains(StandardCopyOption.REPLACE_EXISTING)
+                    && optionSet.contains(StandardCopyOption.ATOMIC_MOVE)
+                    && Files.exists(target)) {
+                    // simulate a file system that can't do atomic move + overwrite
```

Comment:
It's good to cover this for sure, but I think we should cover the real file systems too.